tests/test_covariance_builder.py

```diff
@@ -7,14 +7,63 @@
 import pickle
 import pyccl as ccl
 import pymaster as nmt
-
+from unittest.mock import MagicMock
 from tjpcov.covariance_builder import CovarianceBuilder
 from scipy.linalg import block_diag
 
 INPUT_YML = "./tests/data/conf_covariance_builder_minimal.yaml"
 OUTDIR = "tests/tmp/"
 
 
+def test_no_config_throws():
+    mock_builder = MagicMock(CovarianceBuilder)
```

Comment:
What magic is `MagicMock` doing?

Reply:
I use `unittest.mock` here to test the logic in the base class without needing to instantiate any concrete implementations.  This is preferable because then we aren't implicitly testing something else.

`MagicMock` is a regular `Mock` that also allows you to mock dunder methods (e.g. `__new__`).  In this case the added mocking allows us to directly call the `__init__` function on the base class.